rgw/v2/tests/nfs_ganesha/test_on_nfs_io.py

```diff
@@ -36,162 +48,153 @@ def test_exec(rgw_user_info_file, config):
-            log.info("writable mount point with Pseudo: %s" % mnt_point)
-
-        if io_op_config.get("create", None) is True:
-            do_io = DoIO(nfs_ganesha.rgw_user_info, mnt_point)
-            # base dir creation
-            for bc in range(io_config["basedir_count"]):
-                basedir_name_to_create = utils.gen_bucket_name_from_userid(
-                    nfs_ganesha.rgw_user_info["user_id"], rand_no=bc
-                )
-                log.info("creating basedir with name: %s" % basedir_name_to_create)
-                write = do_io.write("basedir", basedir_name_to_create)
-                if write is False:
-                    raise TestExecError("write failed on mount point")
-                if io_config["subdir_count"] != 0:
-                    for sd in range(io_config["subdir_count"]):
-                        subdir_name_to_create = utils.gen_bucket_name_from_userid(
-                            basedir_name_to_create + ".subdir", rand_no=sd
-                        )
-                        log.info(
-                            "creating subdir with name: %s" % subdir_name_to_create
-                        )
-                        write = do_io.write(
-                            "subdir",
-                            os.path.join(basedir_name_to_create, subdir_name_to_create),
-                        )
-                        if write is False:
-                            raise TestExecError("write failed on mount point")
 
-                if io_config["file_count"] != 0:
-                    for fc in range(io_config["file_count"]):
-                        file_name_to_create = utils.gen_bucket_name_from_userid(
-                            basedir_name_to_create + ".file", rand_no=fc
+    nfs_ganesha = PrepNFSGanesha(rgw_user_info_file=rgw_user_info_file)
+    mounted = nfs_ganesha.initialize()
+    if mounted is False:
+        raise TestExecError("mount failed")
+    log.info("authenticating rgw user")
+    mnt_point = nfs_ganesha.rgw_user_info["nfs_mnt_point"]
+    if (
+        nfs_ganesha.rgw_user_info["nfs_version"] == 4
+        and nfs_ganesha.rgw_user_info["Pseudo"] is not None
+    ):
+        log.info("nfs version: 4")
+        log.info("adding Pseudo path to writable mount point")
+        mnt_point = os.path.join(mnt_point, nfs_ganesha.rgw_user_info["Pseudo"])
+        log.info("writable mount point with Pseudo: %s" % mnt_point)
+
+    if io_op_config.get("create", None) is True:
+        do_io = DoIO(nfs_ganesha.rgw_user_info, mnt_point)
+        # base dir creation
+        for bc in range(io_config["basedir_count"]):
+            basedir_name_to_create = utils.gen_bucket_name_from_userid(
+                nfs_ganesha.rgw_user_info["user_id"], rand_no=bc
+            )
+            log.info("creating basedir with name: %s" % basedir_name_to_create)
+            write = do_io.write("basedir", basedir_name_to_create)
```

Comment:
it makes sense to have an exception in one place. I would suggest making changes in the script rather than in `DoIo()`.  This function is used in many places I believe. if it is decided to make changes in `DoIo`. make sure that rgw tests are not broken.  just a heads up that's all.